File: jar-client/src/test/java/com/beancounter/client/WiringServices.java
package com.beancounter.client;

import static org.assertj.core.api.Assertions.assertThat;

import com.beancounter.client.services.AssetService;
import com.beancounter.client.services.ClientConfig;
import com.beancounter.client.services.ExchangeService;
import com.beancounter.client.services.FxRateService;
import com.beancounter.client.services.PortfolioService;
import com.beancounter.client.services.PriceService;
import com.beancounter.client.services.StaticService;
import com.beancounter.client.services.TrnService;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.autoconfigure.ImportAutoConfiguration;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.cloud.contract.stubrunner.spring.AutoConfigureStubRunner;
import org.springframework.cloud.contract.stubrunner.spring.StubRunnerProperties;

@ImportAutoConfiguration(ClientConfig.class)
@SpringBootTest(classes = ClientConfig.class)
@AutoConfigureStubRunner(
    stubsMode = StubRunnerProperties.StubsMode.LOCAL,
    ids = "org.beancounter:svc-data:+:stubs:10999")
public class WiringServices {
  @Autowired
  private AssetService assetService;
  @Autowired
  private ExchangeService exchangeService;
  @Autowired
  private FxRateService fxRateService;
  @Autowired
  private PortfolioService portfolioService;
  @Autowired
  private PriceService priceService;
  @Autowired
  private StaticService staticService;
  @Autowired
  private TrnService trnService;

  @Test
  void is_Wired() {
    assertThat(assetService).isNotNull();
    assertThat(exchangeService).isNotNull();
    assertThat(fxRateService).isNotNull();
    assertThat(portfolioService).isNotNull();
    assertThat(priceService).isNotNull();
    assertThat(staticService).isNotNull();
    assertThat(trnService).isNotNull();
  }
}

File: jar-client/src/test/kotlin/com/beancounter/client/WiringServices.kt
package com.beancounter.client

import com.beancounter.auth.TokenService
import com.beancounter.client.config.ClientConfig
import com.beancounter.client.ingest.AssetIngestService
import com.beancounter.client.services.PortfolioServiceClient
import com.beancounter.client.services.PriceService
import com.beancounter.client.services.StaticService
import com.beancounter.client.services.TrnService
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.autoconfigure.ImportAutoConfiguration
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.cloud.contract.stubrunner.spring.AutoConfigureStubRunner
import org.springframework.cloud.contract.stubrunner.spring.StubRunnerProperties
import org.springframework.test.context.bean.override.mockito.MockitoBean

/**
 * Simple wiring tests
 */
@ImportAutoConfiguration(ClientConfig::class)
@SpringBootTest(classes = [ClientConfig::class])
@AutoConfigureStubRunner(
    stubsMode = StubRunnerProperties.StubsMode.LOCAL,
    ids = ["org.beancounter:svc-data:0.1.1:stubs:11000"]
)
class WiringServices {
    @Autowired
    private lateinit var assetIngestService: AssetIngestService

    @Autowired
    private lateinit var fxRateService: FxService

    @Autowired
    private lateinit var portfolioService: PortfolioServiceClient

    @Autowired
    private lateinit var priceService: PriceService

    @Autowired
    private lateinit var staticService: StaticService

    @Autowired
    private lateinit var trnService: TrnService

    @MockitoBean
    private lateinit var tokenService: TokenService

    @Test
    fun is_Wired() {
        Assertions.assertThat(assetIngestService).isNotNull
        Assertions.assertThat(fxRateService).isNotNull
        Assertions.assertThat(portfolioService).isNotNull
        Assertions.assertThat(priceService).isNotNull
        Assertions.assertThat(staticService).isNotNull
        Assertions.assertThat(trnService).isNotNull
    }
}